src/mmrelay/matrix/sync_bootstrap.py

```
@@ -0,0 +1,1201 @@
+        if passed_config is not None:
+            facade.config = local_config_dict  # type: ignore[assignment]
+            config_module.relay_config = local_config_dict
+        facade.matrix_homeserver = local_homeserver  # type: ignore[assignment]
+        facade.matrix_access_token = local_access_token  # type: ignore[assignment]
+        facade.bot_user_id = effective_bot_user_id  # type: ignore[assignment]
+        facade.matrix_rooms = local_matrix_rooms  # type: ignore[assignment]
+        facade.matrix_client = client
+        return client
+
+
+async def login_matrix_bot(
```

Ruff (0.15.9)

[warning] Too many return statements (8 > 6)

(PLR0911)